File: docker/analytics/analytics-spark/src/main/java/com/fh/NormalRange.java
package com.fh;

import com.google.gson.Gson;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class NormalRange {

    private final HashMap<String, Range> map = new HashMap<>();

    public class Range {

        public Range(Double max, Double min) {
            this.max = max;
            this.min = min;
        }

        Double max;
        Double min;
    }

    public NormalRange() {
        String json = new Scanner(getClass().getClassLoader().getResourceAsStream("NormalRange.json"), "UTF-8").useDelimiter("\\A").next();
        Gson gson = new Gson();

        System.out.println(json);

        Map<String, Map<String, Double>> map = new HashMap<>();

        map = (Map<String, Map<String, Double>>) gson.fromJson(json, map.getClass());

        map.forEach((key, value) -> {
            this.map.put(key, new Range(value.get("max"), value.get("min")));
        });
    }

    public boolean isInRange(String unit, double value) {
        Range range = map.get(unit);
        if (range == null) return true;

        return range.min <= value && value <= range.max;
    }

}

File: docker/analytics/analytics-kstreams/src/main/kotlin/com/fh/alerts/NormalRange.kt
package com.fh.alerts

import com.google.gson.Gson
import java.util.*

class NormalRange {

    inner class Range(internal var max: Double, internal var min: Double)

    private val map = HashMap<String, Range>()

    init {
        val json = Scanner(javaClass.classLoader.getResourceAsStream("NormalRange.json")!!, "UTF-8").useDelimiter("\\A").next()
        val gson = Gson()

        println(json)

        var map: Map<String, Map<String, Double>> = HashMap()

        map = gson.fromJson(json, map.javaClass)

        map.forEach { key, value ->
            val max = value["max"]
            val min = value["min"]

            if (max == null || min == null){
                throw NoSuchFieldError()
            }

            this.map[key] = Range(max, min)
        }
    }

    fun isInRange(unit: String, value: Double): Boolean {
        val range = map[unit] ?: return true

        return range.min <= value && value <= range.max
    }

}